### PR TITLE
Upgrade to Swift 4.1

### LINKED ISSUE #77: Swift 4 support
Hi, I'm migrating my project to Swift 4 and am getting an error from Carthage when I try to pull in Vinyl saying I'm using an incompatible version of Swift. Any plans to build the Vinyl framework with Swift 4?

### PR DESCRIPTION
**Pull request by wzrad**
**Changes**
- Upgrades Vinyl to Swift 4.1.
- Picks up stalled PR #78. Resolves #77.

Thanks for getting it started @jane-austin.

### CODE REVIEW

File: Vinyl/Extensions/Bundle.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import Foundation
 
 // Idea taken from Venmo/DVR (thanks!)
-func testingBundle() -> Bundle {
+public func testingBundle() -> Bundle {
```
Comment (wzrad) [inline Vinyl/Extensions/Bundle.swift]: This needs to be public to be used as a default argument to a public initializer.

**Comment by wzrad** [comment]
@mluisbrown If you have a moment, do you think you could give this a quick review? I'm also not totally sure what's up with the travis build. It's failing to locate a tvOS sim, but the query didn't change and it looks like it _should_ match one of the available sims.

File: VinylTests/Arbitrary.swift [hunk 2]
```diff
@@ -11,9 +11,9 @@ import SwiftCheck
 
 /// Generates an array of lowercase alphabetic `Character`s.
 let lowerStringGen =
-Gen<Character>.fromElements(in: "a"..."z")
-    .proliferateNonEmpty
-    .map(String.init)
+    Gen<Character>.fromElements(in: "a"..."z")
+        .proliferateNonEmpty
+        .map { String($0) }
```
Comment (mluisbrown) [inline VinylTests/Arbitrary.swift]: Why was this change necessary? Surely `map(String.init)` still works?

File: VinylTests/Arbitrary.swift [hunk 3]
```diff
@@ -78,7 +78,13 @@ let pathParameterGenArray: Gen<[String]> = Gen.sized { sz in
 }
 /// Generates a set of parameters.
 let pathParameterGen: Gen<String> = pathParameterGenArray.map { xs in
-        return xs.reduce("?") { $0 == "?" ? "?" + $1 : $0 + "&" + $1
+    return xs.reduce("?") { (acc, val) in
+        if acc == "?" {
+            return "?" + val
+        }
+        else {
+            return acc + "?" + val
```
Comment (mluisbrown) [inline VinylTests/Arbitrary.swift]: Should be `return acc + "&" + val`, but why did you break this up from the one liner?

**Comment by wzrad** [comment]
I didn't make those changes, I pulled them in from the other, similar, Swift 4.1 PR. I'm closing this though as I no longer intend to see it through, sorry!
